pytorch_to_returnn/torch/nn/functional.py

```diff
@@ -651,60 +651,60 @@ def multi_head_attention_forward(
     assert bias_k is None
     assert bias_v is None
 
-  q = q.contiguous().view(tgt_len, bsz * num_heads, head_dim).transpose(0, 1)
+  # reshape to (bsz, num_heads, tgt_len, head_dim)
+  q = q.contiguous().view(tgt_len, bsz, num_heads, head_dim).transpose(0, 1).transpose(1, 2)
   if k is not None:
-    k = k.contiguous().view(-1, bsz * num_heads, head_dim).transpose(0, 1)
+    k = k.contiguous().view(-1, bsz, num_heads, head_dim).transpose(0, 1).transpose(1, 2)
```

Comment:
Is this not the same as `...transpose(0, 1, 2)` or sth like that? (I'm pretty sure that you can combine it to a single transpose.)

Reply:
It's possible via `permute` but that's not implemented yet and I had some strange issues when I wanted to add it so I used this instead. Of course it would be nicer with `permute` though.

Reply:
Ah ok. Then just leave it for now. (The RETURNN generated code for this is anyway a no-op.)